app.js

```diff
@@ -8,35 +10,127 @@ app.use(logger("dev"));
 app.use(express.static("public"));
 app.use(express.json());
 
-
 // Iteration 1 - Connect to MongoDB
 // DATABASE CONNECTION
-
-
-
+const MONGODB_URI = "mongodb://127.0.0.1:27017/library-mongoose";
+mongoose
+  .connect(MONGODB_URI)
+  .then((x) =>
+    console.log(`Connected to Mongo! Database name:"${x.connections[0].name}"`)
+  )
+  .catch((err) => console.log("Error connecting to mongo", err));
 
 //  Iteration 3 - Create a Book
-
+/* Book.create({
+    title:"Harry Potter and the Deathly Hallows",
+    author:"JK Rowling",
+    pages: 607,
+    cover:'https://upload.wikimedia.org/wikipedia/en/a/a9/Harry_Potter_and_the_Deathly_Hallows.jpg',
+    genres:["Thriller","Mystery"],
+    isBestSeller:true
+})
+.then((res)=>{
+    console.log(res)
+})
+.catch((err)=>{
+    console.log(err)
+})
+ */
 
 //  Iteration 4 - Add Many Books
-
-
-//  Iteration 5 - Get All Recipes
+/* Book.insertMany([
+    {
+      title: "The Great Gatsby",
+      author: "F. Scott Fitzgerald",
+      pages: 180,
+      cover: "https://example.com/greatgatsbycover.jpg",
+      genres: ["Classic", "Fiction"],
+      isBestSeller: true
+    },
+    {
+      title: "To Kill a Mockingbird",
+      author: "Harper Lee",
+      pages: 281,
+      cover: "https://example.com/tokillamockingbirdcover.jpg",
+      genres: ["Classic", "Fiction", "Drama"],
+      isBestSeller: true
+    },
+    {
+      title: "1984",
+      author: "George Orwell",
+      pages: 328,
+      cover: "https://example.com/1984cover.jpg",
+      genres: ["Classic", "Science Fiction", "Dystopian"],
+      isBestSeller: true
+    },
+    {
+      title: "The Hobbit",
+      author: "J.R.R. Tolkien",
+      pages: 300,
+      cover: "https://example.com/thehobbitcover.jpg",
+      genres: ["Fantasy", "Adventure"],
+      isBestSeller: true
+    },
+    {
+      title: "Harry Potter and the Philosopher's Stone",
+      author: "J.K. Rowling",
+      pages: 332,
+      cover: "https://example.com/harrypottercover.jpg",
+      genres: ["Fantasy", "Young Adult"],
+      isBestSeller: true
+    }
+  ])
+  .then((res)=>{
+    console.log(res)
+})
+.catch((err)=>{
+    console.log(err)
+})
+ */
+
+//  Iteration 5 - Get All Books
+
+Book.find()
+.then((res)=>{
+    console.log(res)
+})
+.catch((err)=>{
+    console.log(err)
+})
 
 
 //  Iteration 6 - Get a Single Book
 
+Book.findOne({title:{$regex:"Harry Potter"}})
```

Comment:
ill suggest you to use variables for values in order to make it easy to read and mantain scability applies for all values that are directly pass to the functions , btw nice work with the regex
```suggestion
 const bookTitle = "Harry Potter"
 Book.findOne({ title: { $regex: bookTitle } })
```